### PR TITLE
build: compile and sign sample apps from local machine with fastlane

### PR DESCRIPTION
**Pull request by levibostian**
Part of: [linear.app/customerio/issue/MBL-421](https://linear.app/customerio/issue/MBL-421/)

# Problem 

Part of getting RN SDK CI to build sample apps is to setup the tooling (fastlane) to build and sign iOS sample apps. We need config files setup so you can run `fastlane ios build` and be able to compile, sign, and upload iOS apps builds. 

# Solution 

Re-use the fastlane config files from iOS SDK. This PR adds all the config files so I could build and sign iOS sample apps using fastlane from my local machine. 

This PR is a companion to iOS SDK PR: https://github.com/customerio/customerio-ios/pull/778

Steps to run on your machine: 
* cd Apps/
* bundle install
* cd FCM/ios or APN/ios
* bundle exec fastlane ios build

Follow-up PRs for setting up CI to compile the apps.

### CODE REVIEW

File: Apps/APN/fastlane/Fastfile
Comment (levibostian) [inline Apps/APN/fastlane/Fastfile:1]: File moved into `Apps/APN/ios/` directory. Required so fastlane can find Xcode project files.

File: Apps/APN/ios/fastlane/Fastfile
Comment (levibostian) [inline Apps/APN/ios/fastlane/Fastfile:1]: [File moved, not new](https://github.com/customerio/customerio-reactnative/pull/287/files#r1684833833).

File: Apps/Gemfile
Comment (levibostian) [inline Apps/Gemfile:1]: File moved from `Apps/FCM` directory into `Apps` so it can be re-used by all sample apps.

File: Apps/FCM/Gemfile
Comment (levibostian) [inline Apps/FCM/Gemfile:1]: [File moved, not deleted](https://github.com/customerio/customerio-reactnative/pull/287/files#r1684835239).

File: Apps/FCM/ios/fastlane/Fastfile
Comment (levibostian) [inline Apps/FCM/ios/fastlane/Fastfile:1]: [File moved, not new](https://github.com/customerio/customerio-reactnative/pull/287/files#r1684833833)

File: Apps/fastlane/Fastfile [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+# See the Fastfiles for those projects to learn more about how to use them. 
 
-# Import functions for setting up code signing in development as well as CI server. 
-# See this README (https://github.com/customerio/apple-code-signing) for list of all fastlane functions that this imports and you can run. 
 import_from_git(
-  url: "https://github.com/customerio/apple-code-signing.git", 
-  branch: "main", 
-  path: "fastlane/Fastfile"
+  url: "https://github.com/customerio/customerio-ios.git", 
+  branch: "levi/ci/make-fastlane-reusable",
```
Comment (levibostian) [inline Apps/fastlane/Fastfile:8]: Branch will be updated to `main` in future PR once iOS PRs all merged.

**Review by Shahroz16 — COMMENTED** [review]
Changes look okay to me, but i see Rehans working on the similar CI changes, are they on top of these?

**Comment by levibostian** [comment]
@Shahroz16, Rehan and I have discussed all of the exiting PRs for this ticket and the implementation. So, yes, I do expect that Rehan is building changes on top of this PR. I assume that if implementation goes as planned, there will be more PRs stacked on this one.

**Pull request closed without merging**